### PR TITLE
feat(language): update code example for Cloud NL v2 - python

### PR DESCRIPTION
**Pull request by kornosk**
## Description

Fixes b/296927741

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] These samples need a new **API enabled** in testing projects to pass (let us know which ones): b/295392278.
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: language/v2/language_entities_gcs.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#
+# Copyright 2020 Google LLC
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:2]: The year should reflect when the files were first introduced to the repository.
```suggestion
# Copyright 2023 Google LLC
```

File: language/v2/language_entities_gcs.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+#   description: Analyzing Entities in text file stored in Cloud Storage
+#   usage: python3 samples/v2/language_entities_gcs.py [--gcs_content_uri "gs://cloud-samples-data/language/entity.txt"]
+
+# [START language_entities_gcs]
+from google.cloud import language_v2
+
+
+def sample_analyze_entities(gcs_content_uri):
+    """
+    Analyzing Entities in text file stored in Cloud Storage
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:32]: Use [descriptive style](https://google.github.io/styleguide/pyguide#383-functions-and-methods) for docstrings, as well as using full-sentence format (period at the end, proper grammar, etc)
```suggestion
    Analyzes Entities in text file stored in Cloud Storage.
```

File: language/v2/language_entities_gcs.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+# sample-metadata
+#   title: Analyzing Entities (GCS)
+#   description: Analyzing Entities in text file stored in Cloud Storage
+#   usage: python3 samples/v2/language_entities_gcs.py [--gcs_content_uri "gs://cloud-samples-data/language/entity.txt"]
+
+# [START language_entities_gcs]
+from google.cloud import language_v2
+
+
+def sample_analyze_entities(gcs_content_uri):
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:30]: can you add type hints for the variable and the return type?

File: language/v2/language_entities_gcs.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    """
+    Analyzing Entities in text file stored in Cloud Storage
+
+    Args:
+      gcs_content_uri Google Cloud Storage URI where the file content is located.
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:35]: Please follow the format in https://google.github.io/styleguide/pyguide#383-functions-and-methods:

```
Args:
  gcs_content_uri: ...
```

File: language/v2/language_entities_gcs.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      e.g. gs://[Your Bucket]/[Path to File]
+    """
+
+    client = language_v2.LanguageServiceClient()
+
+    # gcs_content_uri = 'gs://cloud-samples-data/language/entity.txt'
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:41]: Should this be removed?

File: language/v2/language_entities_gcs.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    # Available types: PLAIN_TEXT, HTML
+    type_ = language_v2.Document.Type.PLAIN_TEXT
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:44]: consider a more descriptive name like `document_type_in_plain_text` or `type_in_lain_text`.

File: language/v2/language_entities_gcs.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    # For list of supported languages:
+    # https://cloud.google.com/natural-language/docs/languages
+    language_code = "en"
+    document = {
+        "gcs_content_uri": gcs_content_uri,
+        "type_": type_,
+        "language_code": language_code,
+    }
+
+    # Available values: NONE, UTF8, UTF16, UTF32
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:56]: Is there a link you can provide, so this doesn't become stale? The one you provided above for language is great!

File: language/v2/language_entities_gcs.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+    response = client.analyze_entities(
+        request={"document": document, "encoding_type": encoding_type}
+    )
+
+    # Loop through entitites returned from the API
+    for entity in response.entities:
+        print(f"Representative name for the entity: {entity.name}")
+
+        # Get entity type, e.g. PERSON, LOCATION, ADDRESS, NUMBER, et al
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:67]: Ditto for a more comprehensive list to link to.

File: language/v2/language_entities_gcs.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    # Available values: NONE, UTF8, UTF16, UTF32
+    encoding_type = language_v2.EncodingType.UTF8
+
+    response = client.analyze_entities(
+        request={"document": document, "encoding_type": encoding_type}
+    )
+
+    # Loop through entitites returned from the API
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:63]: nit: comments should explain why the code is written in a certain way or help provide additional context, it shouldn't reiterate the code's behavior in plain English. While this is a sample application and comments help, too much comments can also hinder readability and should be added to a guide rather than in the sample code itself.

File: language/v2/language_entities_gcs.py [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        # The API currently supports proper noun mentions.
+        for mention in entity.mentions:
+            print(f"Mention text: {mention.text.content}")
+
+            # Get the mention type, e.g. PROPER for proper noun
+            print(
+                "Mention type: {}".format(
+                    language_v2.EntityMention.Type(mention.type_).name
+                )
+            )
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:86]: if you chose `format()` type rather than `f-string` for the 80 character limit in the space, use the following:
```suggestion
            print(
                'Mention type:'
                f' {language_v2.EntityMention.Type(mention.type_).name}'
            )
```

File: language/v2/language_entities_text.py [hunk 11]
```diff
@@ -0,0 +1,106 @@
+#
+# Copyright 2020 Google LLC
```
Comment (dandhlee) [inline language/v2/language_entities_text.py:2]: Ditto for license.

File: language/v2/tests/analyzing_entities.test.yaml [hunk 12]
```diff
@@ -0,0 +1,108 @@
+# Copyright 2022 Google LLC
```
Comment (dandhlee) [inline language/v2/tests/analyzing_entities.test.yaml:1]: Ditto.

File: language/v2/language_entities_text.py
Comment (dandhlee) [inline language/v2/language_entities_text.py:1]: Just curious - what's the purpose of this file?

**Review by dandhlee — COMMENTED** [review]
Will you be adding tests?

File: language/v2/language_entities_gcs.py (hunk 5 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:41]: Yes. Removed.

File: language/v2/language_entities_gcs.py (hunk 3 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:30]: This PR should follow the v1 sample here - https://cloud.google.com/natural-language/docs/analyzing-entities#analyzing_entities_in_a_string

File: language/v2/language_entities_text.py (hunk 11 above)
Comment (kornosk) [reply language/v2/language_entities_text.py:2]: Done.

File: language/v2/tests/analyzing_entities.test.yaml (hunk 12 above)
Comment (kornosk) [reply language/v2/tests/analyzing_entities.test.yaml:1]: Done.

File: language/v2/language_entities_text.py
Comment (kornosk) [reply language/v2/language_entities_text.py:1]: This PR publishes the sample code for API v2. See the v1 sample here - https://cloud.google.com/natural-language/docs/analyzing-entities#analyzing_entities_in_a_string

File: language/v2/language_entities_gcs.py (hunk 9 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:61]: Removed.

**Comment by kornosk** [comment]
I copied the test folder from https://github.com/GoogleCloudPlatform/python-docs-samples/tree/HEAD/language/v1/test.

Plan to have the other API samples once this `language_entities` looks good.

File: language/v2/language_entities_gcs.py (hunk 3 above)
Comment (dandhlee) [reply language/v2/language_entities_gcs.py:30]: new samples should adhere to newer style guides, and consistency won't be the best argument for going back on older styles 😓

File: language/v2/language_entities_text.py
Comment (dandhlee) [reply language/v2/language_entities_text.py:1]: Ah, sorry I just wasn't sure since the two modules looked very similar. Got it.

**Comment by dandhlee** [comment]
Gotcha. I'm not entirely sure how the test layout is done since this sample was from a client library repository. I'll leave it to the product team to help verify, but looks like the tests are included in the new run under v2!

File: language/v2/language_entities_gcs.py (hunk 3 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:30]: Agreed. Added the hint types.

File: language/v2/language_entities_gcs.py (hunk 6 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:42]: Done.

File: language/v2/language_entities_gcs.py (hunk 7 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:54]: Added link.

File: language/v2/language_entities_gcs.py (hunk 8 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:64]: Added link.

**Comment by kornosk** [comment]
@dandhlee Thank you for the review! I've addressed all your comments. Please let me know if you have any more concerns.

File: language/v2/language_entities_gcs.py [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+from google.cloud import language_v2
+
+
+def sample_analyze_entities(gcs_content_uri: str) -> None:
+    """
+    Analyzes Entities in text file stored in Cloud Storage.
+
+    Args:
+      gcs_content_uri: Google Cloud Storage URI where the file content is located.
+      e.g. gs://[Your Bucket]/[Path to File]
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:36]:
```suggestion
      gcs_content_uri: Google Cloud Storage URI where the file content is located.
        e.g. gs://[Your Bucket]/[Path to File]
```

File: language/v2/language_entities_gcs.py [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+    # the language specified in the request or, if not specified,
+    # the automatically-detected language.
+    print(f"Language of the text: {response.language_code}")
+
+
+# [END language_entities_gcs]
+
+
+def main():
+    import argparse
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:99]: nit: consider adding the import before the region tag, around line 25 so imports are all in one place - but I'm not 100% sure if the linter would be okay with that. Fine to leave it here for now since this is a more local import.

File: language/v2/language_entities_text.py [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+#   description: Analyzing Entities in a String
+#   usage: python3 samples/v2/language_entities_text.py [--text_content "California is a state."]
+
+# [START language_entities_text]
+from google.cloud import language_v2
+
+
+def sample_analyze_entities(text_content: str) -> None:
+    """
+    Analyzes Entities in a String
```
Comment (dandhlee) [inline language/v2/language_entities_text.py:32]:
```suggestion
    Analyzes Entities in a string.
```

File: language/v2/language_entities_text.py [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+        # The API currently supports proper noun mentions.
+        for mention in entity.mentions:
+            print(f"Mention text: {mention.text.content}")
+
+            # Get the mention type, e.g. PROPER for proper noun
+            print(
+                "Mention type: {}".format(
+                    language_v2.EntityMention.Type(mention.type_).name
+                )
+            )
```
Comment (dandhlee) [inline language/v2/language_entities_text.py:84]:
```suggestion
            print(
                "Mention type:"
                "{language_v2.EntityMention.Type(mention.type_).name}"
            )
```

File: language/v2/language_entities_gcs.py [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+
+        # Loop over the mentions of this entity in the input document.
+        # The API currently supports proper noun mentions.
+        for mention in entity.mentions:
+            print(f"Mention text: {mention.text.content}")
+
+            # Get the mention type, e.g. PROPER for proper noun
+            print(
+                'Mention type:'
+                f' {language_v2.EntityMention.Type(mention.type_).name}'
```
Comment (dandhlee) [inline language/v2/language_entities_gcs.py:83]: Let's be consistent with quote choices (think this is my fault :P)
```suggestion
                "Mention type:"
                f" {language_v2.EntityMention.Type(mention.type_).name}"
```

**Review by dandhlee — APPROVED** [review]
Reviewing for Python style.

File: language/v2/language_entities_gcs.py (hunk 14 above)
Comment (kornosk) [reply language/v2/language_entities_gcs.py:99]: Thanks!

File: language/v2/language_sentiment_text.py [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    Analyzes Sentiment in a string.
+
+    Args:
+      text_content: The text content to analyze.
+    """
+
+    client = language_v2.LanguageServiceClient()
+
+    # text_content = 'I am so happy and joyful.'
```
Comment (dandhlee) [inline language/v2/language_sentiment_text.py:40]: Remove this?

File: language/v2/language_entities_text.py [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+
+        # Loop over the mentions of this entity in the input document.
+        # The API currently supports proper noun mentions.
+        for mention in entity.mentions:
+            print(f"Mention text: {mention.text.content}")
+
+            # Get the mention type, e.g. PROPER for proper noun
+            print(
+                "Mention type:"
+                "{language_v2.EntityMention.Type(mention.type_).name}"
```
[the remark this answers is not in the document]
Comment (dandhlee) [reply language/v2/language_entities_text.py]: Stale comment, but tests not failing even with this wrong formatting is concerning.

File: language/v2/language_classify_gcs.py [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+from google.cloud import language_v2
+
+
+def sample_classify_text(gcs_content_uri: str) -> None:
+    """
+    Classifies Content in text file stored in Cloud Storage
+
+    Args:
+      gcs_content_uri: Google Cloud Storage URI where the file content is located.
+        e.g. gs://[Your Bucket]/[Path to File]. The text file must include at least 20 words.
```
[the remark this answers is not in the document]
Comment (kornosk) [reply language/v2/language_classify_gcs.py:36]: Ahh, I think this is for the previous version. The new version can handle with the input < 20 words now. Thanks for catching this!

Removed the comment.

File: language/v2/language_entities_text.py (hunk 19 above)
[the remark this answers is not in the document]
Comment (kornosk) [reply language/v2/language_entities_text.py]: Yeah, I just fixed this when I was manually checking the code XD.

File: language/v2/language_sentiment_text.py (hunk 18 above)
Comment (kornosk) [reply language/v2/language_sentiment_text.py:40]: Done.

**Comment by telpirion** [comment]
Be sure to review the [Python style guidance](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md) and [Style Guide](https://googlecloudplatform.github.io/samples-style-guide/).

**Comment by kornosk** [comment]
Added tests and minor changes in style.

File: language/v2/language_classify_gcs_test.py [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+import language_classify_gcs
+
+
+def test_sample_classify_text_gcs(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_classify_gcs.sample_classify_text()
+    captured = capsys.readouterr()
+    assert (
+        "TV" in captured.out
```
Comment (telpirion) [inline language/v2/language_classify_gcs_test.py:27]: issue: don't test for model results in these integration tests. Model changes can affect API output, which leads to these tests being broken.

Instead, test for the string literals printed to stdout (e.g. "Confidence" and "Category"). These literals won't be printed if the call to the API fails, which provides sufficient signal for sample quality.

File: language/v2/language_classify_text_test.py [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+import language_classify_text
+
+
+def test_sample_classify_text(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_classify_text.sample_classify_text()
+    captured = capsys.readouterr()
+    assert (
+        "TV" in captured.out
```
Comment (telpirion) [inline language/v2/language_classify_text_test.py:27]: See previous about API output.

File: language/v2/language_entities_gcs_test.py [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def test_sample_analyze_entities_gcs(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_entities_gcs.sample_analyze_entities()
+    captured = capsys.readouterr()
+    assert (
+        "Representative name for the entity: California" in captured.out
+        and "Entity type: LOCATION" in captured.out
```
Comment (telpirion) [inline language/v2/language_entities_gcs_test.py:28]: See previous about testing for API output.

File: language/v2/language_entities_gcs_test.py [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+        and "Mention text: California" in captured.out
```
Comment (telpirion) [inline language/v2/language_entities_gcs_test.py:29]: nit: I would put each of these bool `in` operations into separate asserts & lines.

File: language/v2/language_entities_text.py [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+            print(f"Mention type: {language_v2.EntityMention.Type(mention.type_).name}")
+
+            # Get the probability score associated with the first mention of the entity in the (0, 1.0] range.
+            print(f"Probability score: {mention.probability}")
+
+    # Get the language of the text, which will be the same as
+    # the language specified in the request or, if not specified,
+    # the automatically-detected language.
+    print(f"Language of the text: {response.language_code}")
+
```
Comment (telpirion) [inline language/v2/language_entities_text.py:86]: Remove extraneous line

File: language/v2/language_entities_text_test.py [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+import language_entities_text
+
+
+def test_sample_analyze_entities_text(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_entities_text.sample_analyze_entities()
+    captured = capsys.readouterr()
+    assert (
+        "Representative name for the entity: California" in captured.out
```
Comment (telpirion) [inline language/v2/language_entities_text_test.py:27]: See previous about testing for output.

File: language/v2/language_sentiment_gcs.py [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+    for sentence in response.sentences:
+        print(f"Sentence text: {sentence.text.content}")
+        print(f"Sentence sentiment score: {sentence.sentiment.score}")
+        print(f"Sentence sentiment magnitude: {sentence.sentiment.magnitude}")
+
+    # Get the language of the text, which will be the same as
+    # the language specified in the request or, if not specified,
+    # the automatically-detected language.
+    print(f"Language of the text: {response.language_code}")
+
```
Comment (telpirion) [inline language/v2/language_sentiment_gcs.py:73]: Remove extraneous empty line.

File: language/v2/language_sentiment_gcs_test.py [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+import language_sentiment_gcs
+
+
+def test_sample_analyze_sentiment_gcs(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_sentiment_gcs.sample_analyze_sentiment()
+    captured = capsys.readouterr()
+    assert (
+        "Document sentiment score: 0." in captured.out
```
Comment (telpirion) [inline language/v2/language_sentiment_gcs_test.py:27]: See previous about testing for output.

File: language/v2/language_sentiment_text_test.py [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+import language_sentiment_text
+
+
+def test_sample_analyze_sentiment_text(capsys: ...) -> None:
+    assert os.environ["GOOGLE_CLOUD_PROJECT"] != ""
+
+    language_sentiment_text.sample_analyze_sentiment()
+    captured = capsys.readouterr()
+    assert (
+        "Document sentiment score: 0." in captured.out
```
Comment (telpirion) [inline language/v2/language_sentiment_text_test.py:27]: See previous about testing for API output.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: language/v2/language_classify_gcs.py [hunk 30]
```diff
@@ -0,0 +1,84 @@
...
+        print(f"Category name: {category.name}")
+        # Get the confidence. Number representing how certain the classifier
+        # is that this category represents the provided text.
+        print(f"Confidence: {category.confidence}")
+
+
+# [END language_classify_gcs]
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (kornosk) [reply language/v2/language_classify_gcs.py]: Not sure you still see this but I've removed all the main() functions already. Probably a github glitch or browser not refreshed.

File: language/v2/language_classify_text.py [hunk 31]
```diff
@@ -0,0 +1,85 @@
...
+        print(f"Category name: {category.name}")
+        # Get the confidence. Number representing how certain the classifier
+        # is that this category represents the provided text.
+        print(f"Confidence: {category.confidence}")
+
+
+# [END language_classify_text]
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (kornosk) [reply language/v2/language_classify_text.py]: It's already removed.

File: language/v2/language_classify_gcs_test.py (hunk 21 above)
Comment (kornosk) [reply language/v2/language_classify_gcs_test.py:27]: Done.

File: language/v2/language_entities_text.py (hunk 25 above)
Comment (kornosk) [reply language/v2/language_entities_text.py:86]: Removed.

File: language/v2/language_sentiment_gcs.py (hunk 27 above)
Comment (kornosk) [reply language/v2/language_sentiment_gcs.py:73]: Removed.

File: language/v2/language_sentiment_gcs_test.py (hunk 28 above)
Comment (kornosk) [reply language/v2/language_sentiment_gcs_test.py:27]: Done.

File: language/v2/language_sentiment_text_test.py (hunk 29 above)
Comment (kornosk) [reply language/v2/language_sentiment_text_test.py:27]: Done.

File: language/v2/language_entities_text_test.py (hunk 26 above)
Comment (kornosk) [reply language/v2/language_entities_text_test.py:27]: Done.

File: language/v2/language_entities_gcs_test.py (hunk 24 above)
Comment (kornosk) [reply language/v2/language_entities_gcs_test.py:29]: Good idea! Done.

File: language/v2/language_entities_gcs_test.py (hunk 23 above)
Comment (kornosk) [reply language/v2/language_entities_gcs_test.py:28]: Done.

File: language/v2/language_classify_text_test.py (hunk 22 above)
Comment (kornosk) [reply language/v2/language_classify_text_test.py:27]: Done.

**Review by telpirion — APPROVED** [review]
Looks great! Let's merge it when we can.
